quotefault/__init__.py

```diff
@@ -304,53 +299,60 @@ def review_submit(report_id, result):
     Called when Admin decides on a quote being hidden or kept
     """
     metadata = get_metadata()
-    result = int(result)
-    if metadata['is_admin']:
-        # 1 = Keep, 0 = Hide
-        if result:
-            report = Report.query.get(report_id)
+    report = Report.query.get(report_id)
+    if report and metadata['is_admin']:
+        if result == "keep": 
             report.reviewed = True
             db.session.commit()
             flash("Report Completed: Quote Kept")
-        else:
-            report = Report.query.get(report_id)
+        elif result == "hide":
             report.quote.hidden = True
             report.reviewed = True
             db.session.commit()
             flash("Report Completed: Quote Hidden")
+        else:
+            abort(400)
         return redirect('/review')
-    return redirect('/')
+    elif report:
+        abort(403)
+    abort(400)
+    
 
-@app.route('/hide/<report_id>', methods=['POST'])
+@app.route('/hide/<quote_id>', methods=['POST'])
 @auth.oidc_auth
-def hide(report_id):
+def hide(quote_id):
     """
     Hides a quote
     """
     metadata = get_metadata()
-    quote = Quote.query.get(report_id)
-    if ( metadata['uid'] == quote.speaker
+    quote = Quote.query.get(quote_id)
+    if quote and ( metadata['uid'] == quote.speaker
         or metadata['uid'] == quote.submitter
         or metadata['is_admin'] ):
         quote.hidden = True
         db.session.commit()
         flash("Quote Hidden!")
-    return redirect('/storage')
+        return redirect('/storage')
+    elif quote:
+        abort(403)
+    abort(400)

```

Comment:
Same as above. Reduce nesting, return 403, then 404, then results.
Also, this should be a 404 instead of a 400.